### DISCUSSION TITLE
How to define input star coupler

### DISCUSSION THREAD

**Discussion by AhableelTII** [gdsfactory/gdsfactory#2744] [Q&A]
Hi guys,

i kindly would like help in defining an input star coupler in gdsfactory.  What are the steps? Are there any functions that i should be looking for in the API?

Thanks,

**Comment by fp-matt** [comment]
I'm not familiar with input star couplers, but if you want to take a look at pre-existing components and see if there's anything close to what you are looking for, you can take a look through the [Generic PDK](https://gdsfactory.github.io/gdsfactory/components.html)

**Comment by joamatab** [comment]
Hi,

yes, you can take a look at this [repo](https://github.com/dnrobin/awg-python)
